packages/manager/src/features/IAM/Users/UsersTable/Users.tsx

```diff
@@ -67,6 +73,15 @@ export const UsersLanding = () => {
 
   const numCols = isSmDown ? 2 : numColsLg;
 
+  const queryParams = new URLSearchParams(location.search);
+
+  const handleSearch = (value: string) => {
+    queryParams.set('page', '1');
+
+    history.push({ search: queryParams.toString() });
+    setQuery(value);
+  };
```

Comment:
While were at it, let's remove the `query`/ `setQuery` state and store the search query inside of the query params 

```suggestion
  const handleSearch = (value: string) => {
    queryParams.set('page', '1');
    if (value) {
      queryParams.set('query', value);
    } else {
      queryParams.delete('query');
    }
    history.push({ search: queryParams.toString() });
  };
```

This will allow for a better user experience because the user can bookmark search queries if they want

Reply:
yep, I've updated that logic, thanks